cql3/statements/ks_prop_defs.cc

```diff
@@ -121,7 +121,21 @@ std::optional<unsigned> ks_prop_defs::get_initial_tablets(const sstring& strateg
 
     std::optional<unsigned> ret;
 
-    auto it = tablets_options->find("initial");
+    auto it = tablets_options->find("enabled");
+    if (it != tablets_options->end()) {
+        auto enabled = it->second;
+        tablets_options->erase(it);
+
+        if (enabled == "true") {
+            ret = 0; // even if 'initial' is not set, it'll start with auto-detection
+        } else if (enabled == "false") {
+            goto skip;
```

Comment:
> I think there's a problem here: "enabled=false, initial_tablets=5" will complain that "unrecognized tablets option initial_tablets" because you "skip" over the code below that removes the initial_tablets.

That's deliberate. If user doesn't enable tablets all other options become unrecognized.

>  This will be wrong, and confusing for any user who'll see this message.

What do you suggest? Ignore other options or report back some other message?

Reply:
We have a similar use case with compaction options.
In this case we accept all options, including 'enabled', which can be enabled/disabled later on with yet another ALTER TABLE query, while the other options are persisted.

Reply:
It might not work for initial_tablets. There's no dedicated on/off switch on keyspace_metadata nor there's one in system_schema.scylla_keyspaces table. Only the initial_tablets value and if it's present, tablets are ON, if it's missing tablets are OFF